### PR TITLE
KEA-139 As a user, I want to see a complete listing of announcements so that I can be informed of KEA's happenings

### PR DESCRIPTION
**Pull request by doolri1276**
KEA-139 As a user, I want to see a complete listing of announcements so that I can be informed of KEA's happenings

### CODE REVIEW

File: frontend/src/config/navigation.ts [hunk 1]
```diff
@@ -27,6 +29,7 @@ export const linking: LinkingOptions<RootStackParamList> = {
       Events: "events",
       Announcements: "announcements",
       ContactUs: "contact-us",
+      AnnouncementDetail: "announcements/:id",
```
Comment (doolri1276) [inline frontend/src/config/navigation.ts:32]: @hyunghoseo Would this be the right way to setup path for the AnnouncementDetail Page?

File: frontend/src/components/AnnouncementCard/AnnouncementCard.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  const getPlainText = (rawText: { children: { text: any }[] }[]) => {
+    let text = "";
+    for (const index in rawText) {
+      text += rawText[index].children[0].text + "\n";
+    }
+    return text;
+  };
+
+  console.log(props.id);
```
Comment (doolri1276) [inline frontend/src/components/AnnouncementCard/AnnouncementCard.tsx:34]: Remove log

File: frontend/package.json [hunk 3]
```diff
@@ -69,4 +71,4 @@
     "typescript": "^5.3.0"
   },
   "private": true
-}
\ No newline at end of file
+}
```
Comment (doolri1276) [inline frontend/package.json:74]: Revert

File: frontend/src/config/navigation.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import { LinkingOptions } from "@react-navigation/native";
+import { UndefinedInitialDataInfiniteOptions } from "@tanstack/react-query";
```
Comment (doolri1276) [inline frontend/src/config/navigation.ts:2]: Remove

File: frontend/src/config/navigation.ts [hunk 5]
```diff
@@ -27,6 +29,7 @@ export const linking: LinkingOptions<RootStackParamList> = {
       Events: "events",
       Announcements: "announcements",
       ContactUs: "contact-us",
+      AnnouncementDetails: "announcements/:id",
```
Comment (doolri1276) [inline frontend/src/config/navigation.ts:32]: I wonder why this path is not working

File: frontend/src/screens/Announcements/AnnouncementsScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            announcements={announcements}
+          />
+        </View>
+      </View>
+    </ScreenWrapper>
+  );
+};
+
+const useStyles = () => {
+  const { isMobile } = useResponsiveLayout();
```
Comment (doolri1276) [inline frontend/src/screens/Announcements/AnnouncementsScreen.tsx:45]: If not used, please remove.

File: frontend/src/config/navigation.ts (hunk 1 above)
Comment (doolri1276) [reply frontend/src/config/navigation.ts]: https://github.com/hyunghoseo/nykea-app/pull/64/files#diff-c51c8dcf322cc8722620e6d22f9b2642059abfcae247a0c9a9b789f3257010f7R57

This was due to linkings not being configured correctly.

File: frontend/src/components/AnnouncementCard/AnnouncementCard.tsx (hunk 2 above)
Comment (doolri1276) [reply frontend/src/components/AnnouncementCard/AnnouncementCard.tsx]: Removed

File: frontend/package.json (hunk 3 above)
Comment (doolri1276) [reply frontend/package.json]: Done

File: frontend/src/config/navigation.ts (hunk 5 above)
Comment (doolri1276) [reply frontend/src/config/navigation.ts:31]: Resolved

File: frontend/src/screens/Announcements/AnnouncementsScreen.tsx (hunk 6 above)
Comment (doolri1276) [reply frontend/src/screens/Announcements/AnnouncementsScreen.tsx:45]: Removed

File: frontend/src/components/AnnouncementCard/AnnouncementCard.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
+import { H3, P } from "@expo/html-elements";
+import Moment from "moment";
+import { Skeleton } from "moti/skeleton";
+import { StyleProp, StyleSheet, ViewStyle } from "react-native";
+import { TouchableOpacity } from "react-native-gesture-handler";
+import { Shadow } from "react-native-shadow-2";
+
+import { useNavigationRef } from "@/contexts/NavigationProvider";
+import { useResponsiveLayout } from "@/hooks/useResponsiveLayout";
```
Comment (doolri1276) [inline frontend/src/components/AnnouncementCard/AnnouncementCard.tsx:9]: Remove if not used.

File: frontend/src/screens/Announcements/AnnouncementsScreen.tsx (hunk 6 above)
Comment (doolri1276) [inline frontend/src/screens/Announcements/AnnouncementsScreen.tsx:45]: Remove if not used

File: frontend/package.json [hunk 8]
```diff
@@ -35,6 +35,7 @@
     "expo-status-bar": "~1.11.1",
     "jest": "^29.2.1",
     "jest-expo": "^50.0.2",
+    "moment": "^2.30.1",
```
Comment (jiahn326) [inline frontend/package.json:38]: What is `moment` do in our project?

File: frontend/src/api/apiComponents.ts
Comment (jiahn326) [inline frontend/src/api/apiComponents.ts:1]: Is this file auto-generated?

File: frontend/src/api/apiFetcher.ts [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import { ApiContext } from "./apiContext";
 
-const baseUrl = "https://admin.nykea.org/api";
+const baseUrl = "http://localhost:1338/api"; //LOCAL
+// const baseUrl = "https://admin.nykea.org/api"; //PROD
```
Comment (jiahn326) [inline frontend/src/api/apiFetcher.ts:4]: Should we keep `baseURL` two versions? or is it better keeping only prod version?

File: frontend/src/api/apiSchemas.ts
Comment (jiahn326) [inline frontend/src/api/apiSchemas.ts:1]: I think I need to run the frontend to check those..

File: frontend/src/components/AnnouncementCard/AnnouncementCard.tsx
Comment (jiahn326) [inline frontend/src/components/AnnouncementCard/AnnouncementCard.tsx:1]: I think I need to run the frontend to check those..

File: frontend/src/config/translations.ts [hunk 10]
```diff
@@ -15,7 +15,8 @@ export type TranslationEntryKey =
   | "page.Groups.types.Administrative"
   | "page.Groups.types.Community"
   | "page.Groups.types.Education"
-  | "page.Groups.types.Activities";
+  | "page.Groups.types.Activities"
```
Comment (jiahn326) [inline frontend/src/config/translations.ts:18]: should we put ; in this line?

Comment (jiahn326) [inline frontend/src/config/translations.ts:18]: should we put ; in this line?

File: frontend/package.json (hunk 8 above)
Comment (doolri1276) [reply frontend/package.json:38]: https://github.com/hyunghoseo/nykea-app/pull/64/files#diff-cdeecc7d68e0fb403af97341cb0f958252ab04ef6f6090d993592918e1fc64beR59

moment module allows us to have the date or time in a specific format. 

You can read more about it here : https://momentjs.com

File: frontend/src/config/translations.ts (hunk 10 above)
Comment (doolri1276) [reply frontend/src/config/translations.ts:18]: Duplicated comment

Comment (doolri1276) [reply frontend/src/config/translations.ts:18]: No. One more item is added in line 19. Therefore `;` which signifies the end of the items should be at the end of line 19.

File: frontend/src/api/apiComponents.ts
Comment (doolri1276) [reply frontend/src/api/apiComponents.ts:1]: All the files in api folder will be generated when you call `yarn generate-api` in `package.json`. The files will be generated based on what you have in your backend folders. Therefore, the answer is they are manually auto-generated(?).

File: frontend/src/api/apiFetcher.ts (hunk 9 above)
Comment (doolri1276) [reply frontend/src/api/apiFetcher.ts]: Good Catch!! This is supposed to be PROD. Updated to switch to LOCAL only when needed and not supposed to be commiting.

File: frontend/src/api/apiSchemas.ts [hunk 11]
```diff
@@ -3510,63 +3603,60 @@ export type ListEventComponent = {
   };
 };
 
-export type SupportLocalizationRequest = {
```
[the remark this answers is not in the document]
Comment (doolri1276) [reply frontend/src/api/apiSchemas.ts:3513]: This is removed as I called `yarn generate-api` to reflect the most recent changes in the backend.

File: frontend/src/api/apiComponents.ts
Comment (jiahn326) [reply frontend/src/api/apiComponents.ts:1]: got it, thank you!

File: frontend/src/api/apiSchemas.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (jiahn326) [reply frontend/src/api/apiSchemas.ts:3513]: okay thank you!

File: frontend/src/config/translations.ts (hunk 10 above)
Comment (jiahn326) [reply frontend/src/config/translations.ts:18]: ohhh got it, thank you!

**Pull request merged**